rllib/core/models/specs/checker.py

```diff
@@ -41,47 +41,37 @@ def convert_to_canonical_format(spec: SpecType) -> Union[Spec, SpecDict]:
         - a single constraint. The constraint can be a Spec object, a type, or None.
 
 
-    Examples of canoncial format #1:
+    .. doctest::
 
-    .. code-block:: python
-        spec = ["foo", ("bar", "baz")]
-        output = convert_to_canonical_format(spec)
-        # output = SpecDict({"foo": None, ("bar", "baz"): None})
+        Examples of canoncial format #1:
 
-        spec = {"foo": int, "bar": {"baz": None}}
-        output = convert_to_canonical_format(spec)
-        # output = SpecDict(
-        #   {"foo": TypeSpec(int), "bar": SpecDict({"baz": None})}
-        # )
+        >>> spec = {'foo': int, 'bar': {'baz': None}}
+        >>> convert_to_canonical_format(spec)
+        SpecDict({'foo': TypeSpec(<class 'int'>), 'bar': SpecDict({'baz': None})})
 
-        spec = {"foo": int, "bar": {"baz": str}}
-        output = convert_to_canonical_format(spec)
-        # output = SpecDict(
-        #   {"foo": TypeSpec(int), "bar": SpecDict({"baz": TypeSpec(str)})}
-        # )
+        >>> spec = ['foo', ('bar', 'baz')]
+        >>> convert_to_canonical_format(spec)
+        SpecDict({'foo': None, 'bar': SpecDict({'baz': None})})
 
-        spec = {"foo": int, "bar": {"baz": TensorSpec("b,h", framework="torch")}}
-        output = convert_to_canonical_format(spec)
-        # output = SpecDict(
-        #   {"foo": TypeSpec(int), "bar": SpecDict({"baz": TensorSpec("b,h",
-        framework="torch")})}
-        # )
+        >>> from ray.rllib.core.models.specs.specs_base import TensorSpec
+        >>> spec = {'bar': {'baz': TensorSpec('b,h', framework='torch')}}
+        >>> convert_to_canonical_format(spec)
+        SpecDict({'bar': SpecDict({'baz': TensorSpec(shape=('b', 'h'), dtype=None)})})
```

Comment:
These examples are okay as doctest-style (`>>>`) because they're small and emphasize object reprs